core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -88,7 +88,7 @@ protected void eventually(String message, Condition ec, long timeoutMillis, int
    }
 
    protected void eventually(Condition ec, long timeout, TimeUnit unit) {
-      eventually(null, ec, timeout, 500, unit);
+      eventually(null, ec, unit.toMillis(timeout), 500, TimeUnit.MILLISECONDS);
```

Comment:
Yeah my test kept hanging for what seemed forever, since it wanted to wait for 500 SECONDS :P